### PR TITLE
Jenkins 2.0 support in Hygieia Jenkins plugin

### PR DESCRIPTION
**Pull request by tabladrum**
This is a working version. Wanted to open up for review and suggestions. There are scopes of refactoring (removing code duplications) but will get to that later.

### CODE REVIEW

**Comment by tabladrum** [comment]
@amitmawkin @satishc1 - I have tested this with 2.35 version of Jenkins 2.0 and multiple versions of Sonar (for Sonar publish). Can you test with your version of Jenkins?

File: hygieia-jenkins-plugin/README.md [hunk 1]
```diff
@@ -2,8 +2,7 @@
 
 # Developer instructions
 
-Install Maven and JDK.  This was last built with Maven 3.2.5 and OpenJDK
-1.7.0\_75 
+Install Maven and JDK.  This was last built with Maven 3.2.5 and JDK 1.8.
```
Comment (amitmawkin) [inline hygieia-jenkins-plugin/README.md]: @tabladrum can you see if you can build this with maven 3.3.9 . The rest of the project was built with that version of maven

File: hygieia-jenkins-plugin/src/main/java/hygieia/builder/ArtifactBuilder.java [hunk 2]
```diff
@@ -66,30 +89,44 @@ private void buildArtifacts() {
...
+                bac.getMetadata().put("buildUrl", HygieiaUtils.getBuildUrl(run));
+                bac.getMetadata().put("buildNumber", HygieiaUtils.getBuildNumber(run));
+                bac.getMetadata().put("jobUrl", HygieiaUtils.getJobUrl(run));
+                bac.getMetadata().put("jobName", HygieiaUtils.getJobName(run));
+                bac.getMetadata().put("instanceUrl", HygieiaUtils.getInstanceUrl(run, listener));
+
+
+                if (run instanceof AbstractBuild) {
+                    AbstractBuild abstractBuild = (AbstractBuild) run;
+                    String scmUrl = HygieiaUtils.getScmUrl(abstractBuild, listener);
```
Comment (MarkRx) [inline hygieia-jenkins-plugin/src/main/java/hygieia/builder/ArtifactBuilder.java]: does WorkflowRun not have this information?

Comment (tabladrum) [inline hygieia-jenkins-plugin/src/main/java/hygieia/builder/ArtifactBuilder.java]: Workflow does not have those environment variables. I will have to fill those in in a different way. But, the question is this: How do we handle multiple SCM repo build? Thoughts?

Comment (tabladrum) [inline hygieia-jenkins-plugin/src/main/java/hygieia/builder/ArtifactBuilder.java]: Workflow does not have those environment variables. I will have to fill those in in a different way. But, the question is this: How do we handle multiple SCM repo build? Thoughts?

Comment (MarkRx) [inline hygieia-jenkins-plugin/src/main/java/hygieia/builder/ArtifactBuilder.java]: Right now we don't with the normal jenkins job so you can just use the first one listed. If you feel adventurous though just add a number to the end of each one after the first one : https://github.com/capitalone/Hygieia/issues/1048#issuecomment-268658328

**Pull request merged**